js/bitget3.js

```diff
@@ -0,0 +1,2899 @@
+        //         "filled_amount":"0.0417000000000000",
+        //         "filled_fees":"0.0000834000000000",
+        //         "match_id":"673491702661292033",
+        //         "order_id":"673491720340279296",
+        //         "price":"359.240000000000",
+        //         "source":"接口",
+        //         "symbol":"eth_usdt",
+        //         "type":"buy-market"
+        //     }
+        //
+        //     swap
+        //
+        //     {
+        //         "trade_id":"6667390",
+        //         "symbol":"cmt_btcusdt",
+        //         "order_id":"525946425993854915",
+        //         "price":"9839.00",
+        //         "order_qty":"3466",
+        //         "fee":"-0.0000528407360000",
+        //         "timestamp":"1561121514442",
+        //         "exec_type":"M",
+        //         "side":"3"
+        //     }
+        //
+        let symbol = undefined;
+        const marketId = this.safeString (trade, 'symbol');
+        let base = undefined;
+        let quote = undefined;
+        if (marketId in this.markets_by_id) {
+            market = this.markets_by_id[marketId];
+            symbol = market['symbol'];
+            base = market['base'];
+            quote = market['quote'];
+        } else if (marketId !== undefined) {
+            const parts = marketId.split ('_');
+            const numParts = parts.length;
+            if (numParts === 2) {
+                const [ baseId, quoteId ] = parts;
+                base = this.safeCurrencyCode (baseId);
+                quote = this.safeCurrencyCode (quoteId);
+                symbol = base + '/' + quote;
+            } else {
+                symbol = marketId.toUpperCase ();
+            }
+        }
+        if ((symbol === undefined) && (market !== undefined)) {
+            symbol = market['symbol'];
+            base = market['base'];
+            quote = market['quote'];
+        }
+        let timestamp = this.safeInteger (trade, 'created_at');
+        timestamp = this.safeInteger2 (trade, 'timestamp', 'ts', timestamp);
+        const priceString = this.safeString (trade, 'price');
+        let amountString = this.safeString2 (trade, 'filled_amount', 'order_qty');
+        amountString = this.safeString2 (trade, 'size', 'amount', amountString);
+        let takerOrMaker = this.safeString2 (trade, 'exec_type', 'liquidity');
+        if (takerOrMaker === 'M') {
+            takerOrMaker = 'maker';
+        } else if (takerOrMaker === 'T') {
+            takerOrMaker = 'taker';
+        }
+        const orderType = this.safeString (trade, 'type');
+        let side = undefined;
+        let type = undefined;
+        if (orderType !== undefined) {
+            side = this.safeString (trade, 'type');
+            type = this.parseOrderType (side);
+            side = this.parseOrderSide (side);
+        } else {
+            side = this.safeString2 (trade, 'side', 'direction');
+            type = this.parseOrderType (side);
```

Comment:
I don't think this is correct. You're saying the the trade type is either "sell" or "buy" when the `orderType` is not defined. 

I'd say it's better to leave it undefined than assign it to a semantically wrong value
![image](https://user-images.githubusercontent.com/43336371/152345557-c7856752-383a-4f4d-9d45-5104ae87da43.png)

Reply:
Fixed in PR #11907